cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
Ah ok got that. But is it really necessary, since the for loop also adds constraints with metabolite.id only if they are not in the constraints (see the if/else before setting coefficient_terms)? I'm probably wrong but deleting the first lines in the function adding the metabolite constraints shouldnt change anything. Of course _populate_solver always has to be called with a non-empty reaction list.